src/client.rs

```diff
@@ -58,10 +100,24 @@ impl DatagoClient {
                     encode_images,
                     image_to_rgb8,
                     engine: None,
+                    is_valid: true,
                 }
             }
-            Err(e) => {
-                panic!("Failed to parse config: {e}");
```

Comment:
issue there is that panic! kind of works when the rust binary is called directly (you get a callstack which points to the config parsing), but when it's after being called by python you get nothing but "python interpreter crashed". So new design is to report to stdio, and return an invalid client (which will just only ever serve null samples) so that the python side can handle the error more gracefully